### ISSUE TITLE
Luanti stuck in vsync, with both vsync disabled in settings, and fps cap set to 4000

### ISSUE THREAD

**Issue by ProunceDev** [luanti-org/luanti#15471]
### Luanti version

```true
Luanti 5.10.0 (Windows)
Using LuaJIT 2.1.1725453128
Built by Clang 18.1
Running on Windows/10.0.19041 x86_64
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="."
STATIC_LOCALEDIR="locale"
```

### Irrlicht device

Win32

### Operating system and version

Windows 10 22H2 ( Issue confirmed on Windows 11 23H2 aswell )

### CPU model

Ryzen 5 3600

### GPU model

NVIDIA RTX 3070

### Active renderer

OpenGL 4.6.0

### Summary

Whenever playing newer versions of minetest ( Minetest 5.7.0 > ), the FPS is always capped at vsync, even with system vsync disabled.
![image](https://github.com/user-attachments/assets/3325a35f-0ac2-4b07-a65d-57b3af052642)
![image](https://github.com/user-attachments/assets/69df6027-7ed8-4880-b80f-e57ee8a516dd)

### Steps to reproduce

Boot minetest 5.10.0, disable vsync and set frame rate limit very high, then check FPS, as long as your PC is able to run minetest faster then your monitors refresh rate, you will notice the difference. 

EDIT by SmallJoker:

Last known good: 5.8.0 (IrrlichtMt 1.9.0mt13)
First known bad: 5.9.0

 * https://github.com/minetest/irrlicht/compare/1.9.0mt13...master
 * https://github.com/minetest/minetest/commits/5.9.0/irr

**Comment by ProunceDev** [comment]
Please note this is avoidable by going in nvidia settings and forcing vsync to be disabled system wide.

This should not be the case because it breaks vsync for every other game and app that uses it.

**Comment by SmallJoker** [comment]
Which is the newest Minetest version that still worked correctly?

**Comment by ProunceDev** [comment]
> Which is the newest Minetest version that still worked correctly?

5.8.0

**Comment by sfan5** [comment]
Does this still happen?

**Comment by blaboing** [comment]
> Does this still happen?

Yes, just had to do it again with current 5.15-dev(OpenGL3) on 2 different systems(Win10/11) with Nvidia GPUs.
But instead of changing the global 3d settings in the Nvidia control panel, just adding it to the 3d program settings and disable VSync there.

**Comment by sfan5** [comment]
Vsync is now implemented again, please re-test with a fresh **dev** build.

**Comment by blaboing** [comment]
> Vsync is now implemented again, please re-test with a fresh **dev** build.

works now, fresh build Win10/Nvidia/without extra Nvidia control panel 3d program settings
